nodeadm/internal/configprovider/userdata.go

```diff
@@ -102,6 +134,39 @@ func parseMultipart(userDataReader *multipart.Reader) (*internalapi.NodeConfig,
 		}
 		return config, nil
 	} else {
-		return nil, fmt.Errorf("Could not find NodeConfig within UserData")
+		return nil, fmt.Errorf("could not find NodeConfig within UserData")
+	}
+}
+
+func decodeIfBase64(data []byte) ([]byte, error) {
+	e := base64.StdEncoding
+	maxDecodedLen := e.DecodedLen(len(data))
+	decodedData := make([]byte, maxDecodedLen)
+	decodedLen, err := e.Decode(decodedData, data)
+	if err != nil {
+		return data, nil
```

Comment:
so none of the possible errors here end up mattering?

Reply:
oh i guess it can only return [`CorruptInputError`](https://cs.opensource.google/go/go/+/refs/tags/go1.22.3:src/encoding/base64/base64.go;l=301-305) 🤔